pkg/apis/pubsub/v1beta1/topic_lifecycle.go

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2019 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/apis/duck/v1beta1"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
+)
+
+// GetCondition returns the condition currently associated with the given type,
+// or nil.
+func (ts *TopicStatus) GetCondition(t apis.ConditionType) *apis.Condition {
+	return topicCondSet.Manage(ts).GetCondition(t)
+}
+
+// GetTopLevelCondition returns the top level condition
+func (ts *TopicStatus) GetTopLevelCondition() *apis.Condition {
+	return topicCondSet.Manage(ts).GetTopLevelCondition()
+}
+
+// IsReady returns true if the resource is ready overall.
+func (ts *TopicStatus) IsReady() bool {
+	return topicCondSet.Manage(ts).IsHappy()
+}
+
+// InitializeConditions sets relevant unset conditions to Unknown state.
+func (ts *TopicStatus) InitializeConditions() {
+	topicCondSet.Manage(ts).InitializeConditions()
+}
+
+func (ts *TopicStatus) SetAddress(url *apis.URL) {
+	if ts.Address == nil {
+		ts.Address = &v1beta1.Addressable{}
+	}
+	if url != nil {
+		ts.Address.URL = url
+		topicCondSet.Manage(ts).MarkTrue(TopicConditionAddressable)
+	} else {
+		ts.Address.URL = nil
+		topicCondSet.Manage(ts).MarkFalse(TopicConditionAddressable, "emptyHostname", "hostname is the empty string")
```

Comment:
nit: since hostname is gone, should reword the reason?

Reply:
Good idea. Done. I changed both the v1alpha1 and v1beta1 branches, so that we don't suddenly change error messages when our reconcilers start using v1beta1.